core/src/block_sync.rs

```diff
@@ -312,36 +291,24 @@ pub mod message {
             block_sync: &mut BlockSynchronizer<S, W>,
         ) {
             match self {
-                Message::LatestBlock(LatestBlock { hash, peer_id }) => {
-                    let latest_block_hash = block_sync.wsv.latest_block_hash();
-                    if *hash != latest_block_hash {
-                        Message::GetBlocksAfter(GetBlocksAfter::new(
-                            latest_block_hash,
-                            block_sync.peer_id.clone(),
-                        ))
-                        .send_to(block_sync.broker.clone(), peer_id.clone())
-                        .await;
-                    }
-                }
                 Message::GetBlocksAfter(GetBlocksAfter { hash, peer_id }) => {
                     if block_sync.batch_size == 0 {
                         iroha_logger::warn!(
                             "Error: not sending any blocks as batch_size is equal to zero."
                         );
                         return;
                     }
+                    if *hash == block_sync.wsv.latest_block_hash() {
+                        return;
+                    }
```

Comment:
Do we need to check it? I think `blocks_after` should just return an empty set if its latest block matches this hash?

Reply:
yes, because `blocks_after` compares previous hash with the given hash so it wouldn't match on the latest block and we would log error as if the block's hash is not present in the blockchain